transmogrifier/sources/marc.py

```diff
@@ -1,8 +1,200 @@
 import logging
+
+from bs4 import Tag
+
+import transmogrifier.models as timdex
 from transmogrifier.sources.transformer import Transformer
 
 logger = logging.getLogger(__name__)
 
 
 class Marc(Transformer):
     """Marc transformer."""
+
+    def get_optional_fields(self, xml: Tag) -> dict:
+        """
+        Retrieve optional TIMDEX fields from a MARC XML record.
+
+        Overrides metaclass get_optional_fields() method.
+
+        Args:
+            xml: A BeautifulSoup Tag representing a single MARC XML record.
+        """
+        fields: dict = {}
+
+        # alternate_titles
+        alternate_title_marc_data = {
+            "130": ("adfghklmnoprst", "Main Entry - Uniform Title"),
+            "240": ("adfghklmnoprs", "Uniform Title"),
+            "246": ("abfghinp", "Varying Form of Title"),
+            "730": ("adfghiklmnoprst", "Added Entry - Uniform Title"),
+            "740": ("anp", "Added Entry - Uncontrolled Related/Analytical Title"),
+        }
+        for marc_tag, marc_field_data in alternate_title_marc_data.items():
+            for datafield in xml.find_all("datafield", tag=marc_tag):
+                if alternate_title_value := (
+                    self.create_subfield_value_string_from_datafield(
+                        datafield,
+                        marc_field_data[0],
+                        " ",
+                    )
+                ):
+                    fields.setdefault("alternate_titles", []).append(
+                        timdex.AlternateTitle(
+                            value=alternate_title_value,
+                            kind=marc_field_data[1],
+                        )
+                    )
+
+        # call_numbers
+        call_numbers_marc_data = {
+            "050": ("a", ""),
```

Comment:
I'm assuming based on `mario` that we're just grabbing the classification number but let me know if you want that to change